metadata-ingestion/src/datahub/ingestion/source/sql/clickhouse.py

```diff
@@ -0,0 +1,631 @@
+
+
+class ClickHouseSource(SQLAlchemySource):
+    config: ClickHouseConfig
+
+    def __init__(self, config, ctx):
+        super().__init__(config, ctx, "clickhouse")
+        self._lineage_map: Optional[Dict[str, LineageItem]] = None
+        self._all_tables_set: Optional[Set[str]] = None
+
+    @classmethod
+    def create(cls, config_dict, ctx):
+        config = ClickHouseConfig.parse_obj(config_dict)
+        return cls(config, ctx)
+
+    def get_workunits(self) -> Iterable[Union[MetadataWorkUnit, SqlWorkUnit]]:
+        for wu in super().get_workunits():
+            if (
+                self.config.include_table_lineage
+                and isinstance(wu, SqlWorkUnit)
+                and isinstance(wu.metadata, MetadataChangeEvent)
+                and isinstance(wu.metadata.proposedSnapshot, DatasetSnapshot)
+            ):
+                dataset_snapshot: DatasetSnapshotClass = wu.metadata.proposedSnapshot
+                assert dataset_snapshot
+
+                lineage_mcp, lineage_properties_aspect = self.get_lineage_mcp(
+                    wu.metadata.proposedSnapshot.urn
+                )
+
+                if lineage_mcp is not None:
+                    lineage_wu = MetadataWorkUnit(
+                        id=f"{self.platform}-{lineage_mcp.entityUrn}-{lineage_mcp.aspectName}",
+                        mcp=lineage_mcp,
+                    )
+                    self.report.report_workunit(lineage_wu)
+
+                    yield lineage_wu
+
+                if lineage_properties_aspect:
+                    aspects = dataset_snapshot.aspects
+                    if aspects is None:
+                        aspects = []
+
+                    dataset_properties_aspect: Optional[DatasetPropertiesClass] = None
+
+                    for aspect in aspects:
+                        if isinstance(aspect, DatasetPropertiesClass):
+                            dataset_properties_aspect = aspect
+
+                    if dataset_properties_aspect is None:
+                        dataset_properties_aspect = DatasetPropertiesClass()
+                        aspects.append(dataset_properties_aspect)
+
+                    custom_properties = (
+                        {
+                            **dataset_properties_aspect.customProperties,
+                            **lineage_properties_aspect.customProperties,
+                        }
+                        if dataset_properties_aspect.customProperties
+                        else lineage_properties_aspect.customProperties
+                    )
+                    dataset_properties_aspect.customProperties = custom_properties
+                    dataset_snapshot.aspects = aspects
+
+                    dataset_snapshot.aspects.append(dataset_properties_aspect)
+
+            # Emit the work unit from super.
+            yield wu
+
+    def _get_db_name(self) -> str:
+        return getattr(self.config, "database_alias")
```

Comment:
Is database alias a mandatory config parameter? 
As I can see there is a database property as well. 
Is it possible to set only database and not database_alias?

Reply:
`database` property is used only for the connection.
`database_alias` is used as `platform_instance` and most likely should be replaced to it.
Both parameters are not mandatory.

Reply:
@ne1r0n : I would recommend using the `platform_instance` config going forward and not having `database_alias` option in this connector, now that we have platform instance support in the core classes.

Reply:
@shirshanka I've changed `database_alias` to `platform_instance`